src/gpu/intel/jit/gemm/include/gemmstone/problem.hpp

```diff
@@ -294,6 +294,12 @@ void GEMMProblem::autoTypeConversions(ngen::HW hw, bool systolicAvailable)
 {
     using namespace ngen;
 
+    // Workaround to enable zero-points support with int8 weights in dynamic quant.
+    if ((asPtrDims >= 2 || bsPtrDims >= 2) && aoPtrDims > -1 && Ta_ext.isInt8() && Tb_ext.isInt8() && Tc.isFP()){
+        Ta = Type::f16;
+        Tb = Type::f16;
```

Comment:
Hey Kealan, I don't quite understand this WA -- is there a reason Ta/Tb can't be set in gen_gemm_kernel_t?

Also, it seems you'd want to consult the scale type (e.g. we have bf16 scales).